extensions/wikia/LanguageWikisIndex/LanguageWikisIndexController.class.php

```diff
@@ -8,11 +8,32 @@ public function __construct() {
 	}
 
 	public function index() {
-
 		$this->specialPage->setHeaders();
 
-		$this->response->setTemplateEngine( \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE );
+		Wikia::addAssetsToOutput( 'language_wikis_index_scss' );
+
+		$this->setVal( 'langWikis', WikiFactory::getLanguageWikis() );
+
+		if ( $this->isClosedWiki() ) {
+			$this->setVal( 'intro', $this->msg( 'languagewikisindex-intro-closed' )->escaped() );
+		} else {
+			$this->setVal( 'intro', $this->msg( 'languagewikisindex-intro' )->escaped() );
+		}
 
+		$createNewWikiLink = GlobalTitle::newFromText( 'CreateNewWiki', NS_SPECIAL, Wikia::COMMUNITY_WIKI_ID )->getFullURL();
+
+		$this->setVal( 'cnwLink', $createNewWikiLink );
+
+		$this->setVal( 'links', [
+			'cnw' => $createNewWikiLink,
+			'fandom' => '//fandom.wikia.com',
+			'fandom-university' => GlobalTitle::newFromText( 'FANDOM University', NS_MAIN, Wikia::COMMUNITY_WIKI_ID )->getFullURL(),
+		] );
 	}
 
+	private function isClosedWiki() {
+		global $wgCityId;
+		return $wgCityId !== WikiFactory::LANGUAGE_WIKIS_INDEX &&
```

Comment:
` $wgCityId !== WikiFactory::LANGUAGE_WIKIS_INDEX` duplicates what `isEmptyDomainWithLanguageWikis` does. Still, I don't like the fact that we make an assumption here that if this is not an "empty wiki", it must be a closed one. Do you think a hook is much more expensive?

Reply:
We're not making that assumption, we're checking `WikiFactory::isPublic( $wgCityId )` below to see if it was closed. I only included this so we didn't bother with the `WikiFactory::isPublic( $wgCityId )` if it wasn't closed.

I don't think hooks are more expensive, but they're a more complex flow. All we really want to know is "is there a city_list entry and it's closed" which is what `WikiFactory::isPublic( $wgCityId )` will check, using a hook to lead to the closed wiki handler makes following the code path more complex as you have to see where the hook is used, what it's changing, etc. without benefit over `WikiFactory::isPublic( $wgCityId )` in my opinion.